napari/_qt/utils.py

```diff
@@ -268,3 +269,18 @@ def delete_qapp(app):
     # calling a second time is necessary on PySide2...
     # see: https://bugreports.qt.io/browse/PYSIDE-1470
     QApplication.instance()
+
+
+def get_viewer_instance():
+    """Inspect QApplication widgets and return instance of QtViewer
+
+    Returns
+    -------
+    QtViewer
+        napari QtViewer instance
+    """
+    # FIXME: this does not work with multiple viewers.
+    for wdg in QApplication.topLevelWidgets():
+        if isinstance(wdg, QMainWindow):
+            return wdg.centralWidget().children()[1]
+    return None  # TODO: raise error here?
```

Comment:
Do you want to assert what you are returning is the right type in case in the childrens of the central window change   and it's not the one at index 1 anymore ?

Reply:
Thinking at loud, especially since you made a comment about multiple viewer. 
Could/should this be a class method that returns the current instance ? like a `Viewer.get_active_viewer()` or similar, instead of a standalone function. 

I don't completely like it as we would need to make the viewer knowledgeable of the Main window ?
It's still something we can do later anyway.

Reply:
Should clarify, this is @tlambert03 's code and we had some discussion around it. I've pulled it out for now but I think Talley wanted to make a separate PR to address this functionality because as you say, this function is somewhat problematic.

Reply:
Resolved by #2638